src/meta/src/manager/catalog/mod.rs

```diff
@@ -1857,6 +1862,157 @@ impl CatalogManager {
         Ok(version)
     }
 
+    pub async fn alter_set_schema(
+        &self,
+        object: alter_set_schema_request::Object,
+        new_schema_id: SchemaId,
+    ) -> MetaResult<NotificationVersion> {
+        let core = &mut *self.core.lock().await;
+        let database_core = &mut core.database;
+
+        database_core.ensure_schema_id(new_schema_id)?;
+        let database_id = database_core
+            .schemas
+            .get(&new_schema_id)
+            .unwrap()
+            .get_database_id();
+
+        let mut relation_infos = Vec::new();
+        let mut objects_to_alter = vec![object];
+        while let Some(object) = objects_to_alter.pop() {
```

Comment:
Can we simply match the target object and handle it manually? It appears that only the associated source and indexes for the table will be altered cascade. Additionally, committing them separately may result in inconsistency if any failover occurs.